quadtree_map.py

```diff
@@ -0,0 +1,303 @@
+from random import randint
+from cartesian_coordinate import CartesianCoordinate
+from mapping_interface import MapInterface
+from mapping_interface import PointState
+
+"""
+@package Quadtree
+This module is the implementation of mapping using quadtree algorithm.
+"""
+
+
+class rectangle():
+    """
+    @brief A rectangle that contans the size of the root/tree/subtree.
+    """
+
+    def __init__(self, x, y, w, h):
+        """
+        @brief The constructor.
+        @param x The x coordinate of the middle point of the rectangle(width).
+        @param y The x coordinate of the middle point of the rectangle(height).
+        @param w The width of the rectangle(left and right from the middle point).
+        @param h The height of the rectangle(above and below from the middle point).
+        """
+
+        self.x = x
+        self.y = y
+        self.width = w
+        self.height = h
+
+    def contains(self, point):
+        """
+        @brief A function to check if a point position is inside the rectangle area.
+        @return Boolean.
+        """
+
+        return (point.x >= self.x - self.width
```

Comment:
Are you sure `point.x >= self.x - self.width` and `and point.y >= self.y - self.height` are correct? I think they should be `point.x >= self.x` and `point.y >= self.y`

Reply:
Yes it is, because self.x and self.y are the middle point of the rectangle, by subbing it with the width/height we get the real boundary/reach/size of the rectangle.